pkg/middlewares/metrics/metrics.go

```diff
@@ -131,6 +144,84 @@ func getMethod(r *http.Request) string {
 	return r.Method
 }
 
+func getRequestTLSVersion(req *http.Request) string {
+	var tlsVersion string
+
+	switch req.TLS.Version {
+	case tls.VersionTLS10:
+		tlsVersion = "TLS 1.0"
+	case tls.VersionTLS11:
+		tlsVersion = "TLS 1.1"
+	case tls.VersionTLS12:
+		tlsVersion = "TLS 1.2"
+	case tls.VersionTLS13:
+		tlsVersion = "TLS 1.3"
+	case tls.VersionSSL30: //nolint
+		tlsVersion = "SSLv3"
+	default:
+		tlsVersion = "unknown"
+	}
+
+	return tlsVersion
+}
+
+func getRequestTLSCipher(req *http.Request) string {
+	switch req.TLS.CipherSuite {
+	case tls.TLS_RSA_WITH_RC4_128_SHA:
```

Comment:
on second thought here, we already have all of the supported suites and their string equivalents stored in https://github.com/containous/traefik/blob/master/pkg/tls/certificate.go#L35, don't you think you could find the value in the map and return the associated string key? That would prevent having to add/update this cipher list if the certificate.go suite list is updated.

This would also apply to the version above.

Does that seem simpler and more extensible?

Reply:
I'm not comfortable to loop over `pkg/tls.CipherSuites` at each request to find the string that correspond to its uint16 code.

So I made a reverse map and a test that ensure that both maps are always in sync.

What do you think ?